### PR TITLE
[MXNET-582] Fix flaky test test_operator_gpu.test_batchnorm_with_type (follow-up)

### PR DESCRIPTION
**Pull request by hcho3**
## Description ##
Addresses #10087. This is a follow-up of #11396, which did not address the flakiness of `ctx_list_v2_3D`.

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [x] Changes are complete (i.e. I finished coding on this PR)

### CODE REVIEW

**Comment by hcho3** [comment]
Currently running 100000 repetitions, to be extra safe.

**Comment by hcho3** [comment]
The latest version survived 100000 repetitions.

File: tests/python/gpu/test_operator_gpu.py [hunk 1]
```diff
@@ -290,12 +289,12 @@ def test_batchnorm_with_type():
...
-    {'ctx': mx.cpu(0), 'norm_data': (4, 2, 3, 5, 5), 'type_dict': {'norm_data': np.float64}},
-    {'ctx': mx.gpu(0), 'norm_data': (4, 2, 3, 5, 5), 'type_dict': {'norm_data': np.float16}},
-    {'ctx': mx.gpu(0), 'norm_data': (4, 2, 3, 5, 5), 'type_dict': {'norm_data': np.float32}},
-    {'ctx': mx.gpu(0), 'norm_data': (4, 2, 3, 5, 5), 'type_dict': {'norm_data': np.float64}}
+    {'ctx': mx.cpu(0), 'norm_data': (3, 2, 3, 2, 3), 'type_dict': {'norm_data': np.float16}},
+    {'ctx': mx.cpu(0), 'norm_data': (3, 2, 3, 2, 3), 'type_dict': {'norm_data': np.float32}},
+    {'ctx': mx.cpu(0), 'norm_data': (3, 2, 3, 2, 3), 'type_dict': {'norm_data': np.float64}},
+    {'ctx': mx.gpu(0), 'norm_data': (3, 2, 3, 2, 3), 'type_dict': {'norm_data': np.float16}},
+    {'ctx': mx.gpu(0), 'norm_data': (3, 2, 3, 2, 3), 'type_dict': {'norm_data': np.float32}},
+    {'ctx': mx.gpu(0), 'norm_data': (3, 2, 3, 2, 3), 'type_dict': {'norm_data': np.float64}}
```
Comment (haojin2) [inline tests/python/gpu/test_operator_gpu.py]: Why are we reducing the sizes further here?

Comment (hcho3) [reply tests/python/gpu/test_operator_gpu.py]: See https://github.com/apache/incubator-mxnet/issues/9916#issuecomment-371736378

Comment (hcho3) [reply tests/python/gpu/test_operator_gpu.py]: Also, notice that #11396 did not modify `ctx_list_v2_3D`.

Comment (haojin2) [reply tests/python/gpu/test_operator_gpu.py]: Just got a chance to take a look at your reply, I'll dig into the 2 links you provided. Thanks.

**Comment by lupesko** [comment]
@haojin2 bouncing this again for you to follow up with the review

**Comment by lupesko** [comment]
@haojin2 kindly bouncing again...

**Pull request merged**